rust/src/lib.rs

```diff
@@ -69,16 +64,17 @@ pub struct Anime {
     pub source_lens: Vec<f64>,
     pub target_tree: TargetTree,
     pub target_lens: Vec<f64>,
-    pub matches: OnceCell<MatchesMap>,
+    pub matches: MatchesMap,
 }
 
 impl Anime {
-    /// Load source and target `LineString` geometries
+    /// Load source and target `LineString` geometries, then finds candidate matches between
+    /// sources and targets.
     ///
     /// This creates two R* Trees using cached envelopes for each component
     /// line in a LineString. In addition to the envelope, the slope and
     /// index of the LineString is stored.
-    pub fn load_geometries(
+    pub fn new(
```

Comment:
I think it's a simpler API to combine these steps. It removes the possibility for the user to try and call interpolation methods before finding matches.

Reply:
Rather than remove this, it makes sense to add a new method, not rename and change the existing one.